cores/esp8266/heap.cpp

```diff
@@ -384,7 +384,62 @@ void IRAM_ATTR vPortFree(void *ptr, const char* file, int line)
     // correct context. umm_malloc free internally determines the correct heap.
     HeapSelectDram ephemeral;
 #endif
-    return heap_vPortFree(ptr,  file, line);
+    return heap_vPortFree(ptr, file, line);
 }
 
+#if (NONOSDK >= (0x30000))
+////////////////////////////////////////////////////////////////////////////////
+/*
+  New for NON-OS SDK 3.0.0 and up
+  Needed for WPA2 Enterprise support. This was not present in SDK pre 3.0
+
+  The NON-OS SDK 3.0.x has breaking changes to pvPortMalloc. They added one more
+  argument for selecting a heap. To avoid breaking the build, I renamed their
+  broken version pvEsprMalloc. To be used, the LIBS need to be edited.
+
```

Comment:
btw, `pvPortMalloc` -> `sdk3_pvPortMalloc`? (or just `sdk_`)
we can explicitly check for weak symbols, too (either `weakref` attr or just `if (FUNC_NAME)`)

(also just noticed the https://github.com/espressif/ESP8266_NONOS_SDK/blob/158bb7a53f16cfff21b35fb4ec66fa15261f5a4a/include/mem.h)

Reply:
For binary file editing, I need to keep the length constant. `pvSdk3Malloc`  would work.

New `mem.h` - I think it would work after a global replace on `pvPortMalloc`. It might make porting things around easier.

Reply:
```
> xtensa-lx106-elf-objcopy --help | grep redefine
     --redefine-sym <old>=<new>    Redefine symbol name <old> to <new>
     --redefine-syms <file>        --redefine-sym for all symbol pairs
```
?
(even works with archives)

Reply:
I gave that a try. It worked on 33 out of 34 values.  Very strange, I ran the script a second time and it picked up the one it missed. When I kept the string length the same on each side of the "=" it didn't miss any. I would rather use obcopy instead of a brut force binary edit. Are you good with `pvSdk3Malloc` instead of `sdk3_pvPortMalloc`?

Reply:
Weird. Quick test for 305
```
~/.p/p/f/t/s/l/NONOSDK305 (master)> cat names.sh
git checkout -f ./

echo "before:"
xtensa-lx106-elf-nm *.a | grep pvPort | sort -u

for objfile in *.a ; do
    xtensa-lx106-elf-objcopy \
        --redefine-sym pvPortMalloc=sdk3_pvPortMalloc \
        --redefine-sym pvPortRealloc=sdk3_pvPortRealloc \
        --redefine-sym pvPortZalloc=sdk3_pvPortZalloc \
        --redefine-sym pvPortZallocIram=sdk3_pvPortZallocIram \
        $objfile
done

echo "after:"
xtensa-lx106-elf-nm *.a | grep pvPort | sort -u
~/.p/p/f/t/s/l/NONOSDK305 (master)> bash names.sh
Updated 11 paths from the index
before:
         U pvPortMalloc
         U pvPortRealloc
         U pvPortZalloc
         U pvPortZallocIram
after:
         U sdk3_pvPortMalloc
         U sdk3_pvPortRealloc
         U sdk3_pvPortZalloc
         U sdk3_pvPortZallocIram
```

Main idea right now is to have distinct signatures for sdk2 and sdk3 so we don't run into any accidents

Reply:
Thanks! I got it resolved. The missed symbol was temporarily in a .o file and was merged back into the .a after they had been renamed. I reorganized a few things and it is working.